clarity/src/vm/tests/contracts.rs

```diff
@@ -889,7 +926,14 @@ fn test_aborts(owned_env: &mut OwnedEnvironment) {
     );
 }
 
-fn test_factorial_contract(owned_env: &mut OwnedEnvironment) {
+#[apply(test_clarity_versions)]
+fn test_factorial_contract(
+    version: ClarityVersion,
+    epoch: StacksEpochId,
+    mut env_factory: MemoryEnvironmentGenerator,
+) {
+    let mut owned_env = env_factory.get_env(epoch);
+
     let mut placeholder_context = ContractContext::new(
         QualifiedContractIdentifier::transient(),
         ClarityVersion::Clarity2,
```

Comment:
Here and several other places in this file, there are tests where you've added `test_clarity_versions` but then didn't use `version`.

Reply:
Corrected in [7c10037](https://github.com/stacks-network/stacks-blockchain/pull/3731/commits/7c10037a7d539fb9f1cc4f9da990cd74fc115d3b). I added a `warn(unused)` lint definition to the tests module, and fixed all those warnings, then removed the label again, because it interacts pretty unsatisfactorily with the workspace build (it warns on unused items when the `testing` build flag is set, which generates a ton of invalid warns -- there's probably a way to fix that, but I'll save that for another time).